### ISSUE TITLE
roadId 15 crash placing road vehicles and trams on curves

### ISSUE THREAD

**Issue by LeeSpork** [OpenLoco/OpenLoco#2751]
**Version information:**
 - Platform: Windows 10
 - Version: 24.10
 - Commit: 17e974288872f9e3a64f9a96c157cf32fea0b38d

**Describe the bug**
When selecting a location to place a road vehicle or tram, the game can crash if you hover your mouse over a curve.

**To Reproduce**
1. Play any scenario
2. Select any type of road or tram track
3. Select either curve size (smaller curves seem easier for reproducing)
4. Build a circle
5. Buy a bus, truck, or tram
6. While in the mode where you are placing the vehicle, wave your mouse over the curves until the game crashes.

**Additional context**
Crash is an array subscript out of range error, due to the value of `roadId` in `Vehicle1::updateRoad()` becoming 15.

**Comment by AaronVanGeffen** [comment]
Does this still happen on master?

**Comment by LeeSpork** [comment]
I was on this branch in the 3rd screenshot: https://github.com/LeeSpork/OpenLoco/tree/actually-fix-single-music

**Comment by LeeSpork** [comment]
Edited issue after finding simpler reproduction steps.

**Comment by duncanspumpkin** [comment]
Seeing as the are only 10 valid roadids i could see why this would crash.

**Comment by LeeSpork** [comment]
I could not reproduce in original Locomotion

**Comment by shusaura85** [comment]
Can't reproduce it.
@LeeSpork can you confirm if this issue is still present. If not, the issue can be closed.

**Comment by LeeSpork** [comment]
Yup cannot reproduce in newer versions. May have been fixed in v25.09
